docs/src/main/docs/guides/91_mp-tutorial.adoc

```diff
@@ -0,0 +1,1258 @@
+    }
+
+}
+---- 
+
+<1> This updated class adds a `GreetingProvider` and uses constructor injection
+ to get the value from the configuration file.  
+<2> The logic to create the 
+ response message is refactored into a `createResponse` method and the 
+ `getDefaultMessage()` method is updated to use this new method.
+<3> In `createResponse()`, the message is obtained from the `GreetingProvider`
+ which in turn got it from the configuration. 
+
+Rebuild and run the application and notice that it now uses the greeting
+ from the configuration file.  Change the configuration file and restart
+ the application, notice that it uses the changed value.
+
+==== Dynamic configuration
+
+Helidon also allows applications to consume configuration from multiple
+ sources and to consume changes to the configuration dynamically, without requiring a restart. Update the main class as follows:
+
+[source,java] 
+.Updated main class
+----
+package io.helidon.examples;
+
+import io.helidon.config.Config;
+import io.helidon.config.PollingStrategies;
+import io.helidon.microprofile.server.Server;
+
+import java.io.IOException;
+
+import static io.helidon.config.ConfigSources.classpath;
+import static io.helidon.config.ConfigSources.file;
+
+public final class Main {
+
+    private Main() { }
+
+    public static void main(final String[] args) throws IOException {
+        Server server = startServer();
+        System.out.println("http://localhost:" + server.port() + "/greet");
+    }
+
+    private static Config buildConfig() {
+        return Config.builder()
+                .sources(
+                        file("conf/mp.yaml") <1>
+                            .pollingStrategy(PollingStrategies::watch)
+                            .optional(),
+                        classpath("META-INF/microprofile-config.properties"))
+                .build();
+    }
+
+    static Server startServer() {
+        return Server.builder()
+                     .config(buildConfig()) <2>
+                     .build()
+                     .start();
+    }
+
+}
+----
+
+These updates introduce a new configuration file `conf/mp.yaml` which 
+ can be kept outside of the JAR file, and which can be used to provide
+ additional configuration data and/or to override the data in the
+ `microprofile-config.properties` which will be packaged in the JAR file.
+
+Layered configuration like this, where the configuration is read from 
+ more than one source with overrides, is commonly used in cases where 
+ you want to have different configurations for different environments
+ like development, test and production.
+
+TIP: Learn much more about configuration in <<config/06_advanced-configuration.adoc,Advanced Configurtion Topics>>.
+
+Create this file with the following content: 
+
+[source,yaml]
+.Initial mp.yaml configuration file
+----
+app:
+  greeting: "Hallo"
+----
+
+<1> The new `buildConfig()` method defines two sources for configuration information, first the
```

Comment:
This is not how it works - the configuration is a snapshot (immutable). 
You need to react to configuration changes using the `onChange` method on a configuration node.
We have a pull request for that feature, not yet working.

Reply:
should i just remove the dynamic config part until that PR is merged and gets into a release?

Reply:
i removed it